### ISSUE TITLE
Louvain doesnt seem to ever finish on a small graph

### ISSUE THREAD

**Issue by tomasonjo** [neo4j-contrib/neo4j-graph-algorithms#398]
I tried it with big graph and it never finished, so I tried it on a small graph with 612 edges:

Create small graph:

```
:play movies
```
Create network:
```
MATCH (p1:Person)-->(:Movie)<--(p2:Person) where id(p1) < id(p2)
MERGE (p1)-[r:KNOWS]-(p2)
ON CREATE SET r.weight = 1
ON MATCH SET r.weight = r.weight + 1
```

Run the algo without weights:

```
CALL algo.clustering.louvain('Person', 'KNOWS', {write:true, partitionProperty:'cluster', concurrency:2}) 
                YIELD nodes, communityCount, iterations, loadMillis, computeMillis, writeMillis
```

I get nothing in the logs and this does not finish after an hour...

**Comment by mknblch** [comment]
Could you please try again using a weight? 

`CALL algo.clustering.louvain('Person', 'KNOWS', {property:'weight', write:true, partitionProperty:'cluster', concurrency:2}) ...`

**Comment by tomasonjo** [comment]
Tried both property and weightProperty... still does not finish

**Comment by mknblch** [comment]
Ok seems like the algorithm does not converge. I'll add support for an optional `iterations` parameter in the config as a quick fix. So we can test the algo. But it would still be nice to know why the algorithm does not converge on the given graph. Im going to check this

**Comment by mknblch** [comment]
should have been fixed in #405
